packages/nodes-base/nodes/Google/Sheet/v2/actions/sheet/append.operation.ts

```diff
@@ -226,7 +226,9 @@ export async function execute(
 		keyRowIndex = locationDefine.headerRow as number;
 	}
 
-	const sheetData = await sheet.getData(range, 'FORMATTED_VALUE');
+	const [sheetNameForKeyRow] = range.split('!');
```

Comment:
Extracting the sheets name from the "range" is just for the case, that the range contains range annotations.

But the method caller only provides the sheet name as "range" - see [here](https://github.com/n8n-io/n8n/blob/4f27b39b45b58779d363980241e6e5e11b58f5da/packages/nodes-base/nodes/Google/Sheet/v2/actions/router.ts#L63)

Reply:
Another solution could be to check if the range contains a range annotation and only add the keyRowIndex as range annotation if there was no real range provided.